src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java

```diff
@@ -51,17 +51,36 @@
 @SpringBootTest
 public class BlobProcessorTaskTest extends ProcessorTestSuite<BlobProcessorTask> {
 
+    protected static final String VALID_SIGNED_ZIP_FILE_WITH_CASE_NUMBER = "42_24-06-2018-00-00-00.test.zip";
+
     @Before
     public void setUp() throws Exception {
         super.setUp(BlobProcessorTask::new);
     }
 
     @Test
-    public void should_read_from_blob_storage_and_save_metadata_in_database_when_zip_contains_metadata_and_pdfs()
+    public void should_read_blob_and_save_metadata_in_database_when_zip_contains_metadata_and_pdfs()
         throws Exception {
         //Given
         uploadZipToBlobStore(VALID_ZIP_FILE_WITH_CASE_NUMBER); //Zip file with metadata and pdfs
+        testBlobFileProcessed();
+    }
+
+    @Test
+    public void should_read_blob_check_signature_and_save_metadata_in_database_when_zip_contains_metadata_and_pdfs()
+        throws Exception {
+        //Given
+        processor.signatureAlg = "sha256withrsa";
+        processor.publicKeyBase64 =
+            "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDEXfjyDFFigzsmFvTe2cWZ45ggH/XoS/3C6Ur/"
+                + "V0egi8k5hnIIgPEOUqhrX5UcQorSX7bIlMped6TtPkYdGs/QI6S5m2uz+6Mjai7ZfACGhYxIs8"
+                + "35msqvRsDM0tIle/h3eZJb7iPE0anMWb8MkBYU3D3vAnPdBZxiEIwNMUNzqQIDAQAB";
```

Comment:
Looks like a magic string... I assume this is the public key corresponding to the private key with which `42_24-06-2018-00-00-00.test.zip` has been signed.

Can this be made clearer?
Perhaps both the zip file and public key could be put in the same folder in resources?

Reply:
Also, looks like this string is used twice in this class.

Reply:
I've moved the public key to a file